go/vt/vttablet/tabletserver/connpool/dbconn.go

```diff
@@ -132,6 +132,16 @@ func (dbc *DBConn) execOnce(ctx context.Context, query string, maxrows int, want
 	dbc.current.Set(query)
 	defer dbc.current.Set("")
 
+	// Check if the context is already past its deadline before
+	// trying to execute the query.
+	if ctx.Done() != nil {
```

Comment:
This check is not needed. If the channel is nil, it won't fire.

Reply:
Thanks -- removed (and verified it still works)